### PR TITLE
Convert hui-icon-element to TypeScript/LitElement

### PR DESCRIPTION
**Pull request by iantrich**
extract ElementClick mixin functions

- `computeTooltip` only works when called directly in `hui-icon-element` but doesn't when I try to replace what is in the existing `ElementClick` mixin.
- waiting on new navigate function to finish extracting handleClick function and creation of service data needs to be cleaned up for TS linter to like it.

### CODE REVIEW

File: src/common/dom/handle-click.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    case "navigate":
+      // this.navigate(config.navigation_path); // TODO wait for balloob's navigate function
+      break;
+    case "toggle":
+      toggleEntity(hass, config.entity);
+      break;
+    case "call-service": {
+      if (config.service) {
+        const [domain, service] = config.service.split(".", 2);
+        const serviceData = Object.assign(
```
Comment (zsarnett) [inline src/common/dom/handle-click.ts]: Checkout this notation `{entity_id: config.entity, ...config.service_data}`

File: src/common/dom/handle-click.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  if (hold && config.hold_action) {
+    action = config.hold_action;
+  }
+
+  if (action === "none") {
+    return;
+  }
+
+  switch (action) {
```
Comment (zsarnett) [inline src/common/dom/handle-click.ts]: Should there be a default case?

Comment (iantrich) [reply src/common/dom/handle-click.ts]: more-info?

Comment (zsarnett) [reply src/common/dom/handle-click.ts]: Bingo

Comment (iantrich) [reply src/common/dom/handle-click.ts]: This is actually already handled when action is set:

`let action = config.tap_action || "more-info";`

**Comment by iantrich** [comment]
This should be good to go now, I believe

**Comment by iantrich** [comment]
Scratch that. Long press is not working

**Comment by iantrich** [comment]
Scratch the scrath :) I didn't realize that long press was for mobile online. The timing is quite finicky to test.

**Review by zsarnett — APPROVED** [review]

File: src/common/dom/handle-click.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { HomeAssistant } from "../../types";
+import { LovelaceElementConfig } from "../../panels/lovelace/elements/types";
+import { fireEvent } from "../dom/fire_event.js";
+import { navigate } from "../../common/navigate";
+import toggleEntity from "../../../src/panels/lovelace/common/entity/toggle-entity";
+
+export const handleClick = (
+  node: HTMLElement,
+  hass: HomeAssistant,
+  config: LovelaceElementConfig,
```
Comment (balloob) [inline src/common/dom/handle-click.ts]: This belongs in lovelace element folder. Nothing common about it.

File: src/common/string/compute-tooltip.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import computeStateName from "../../common/entity/compute_state_name";
+import { HomeAssistant } from "../../types";
+import { LovelaceElementConfig } from "../../panels/lovelace/elements/types";
+
+export const computeTooltip = (
+  hass: HomeAssistant,
+  config: LovelaceElementConfig
```
Comment (balloob) [inline src/common/string/compute-tooltip.ts]: Same.

**Pull request merged**